packages/testcontainers/src/containers/Hydra/NativeChainContainer.ts

```diff
@@ -0,0 +1,225 @@
+import {
+  GenericContainer,
+  Network,
+  StartedTestContainer
+} from 'testcontainers'
+import fetch from 'cross-fetch'
+import { AbstractStartedContainer } from 'testcontainers/dist/modules/abstract-started-container'
+import { getNetwork, Network as JellyfishNetwork, NetworkName } from '@defichain/jellyfish-network'
+import { JsonRpcClient } from '@defichain/jellyfish-api-jsonrpc'
+
+/**
+ * Mandatory options to start NativeChain with
+ */
+export interface StartOptions {
+  // TODO(fuxingloh): change to cookie based auth soon
+  user: string
+  password: string
+  timeout?: number
+  startFlags?: StartFlags[]
+}
+
+export interface StartFlags {
+  name: string
+  value: number
+}
+
+/**
+ * DeFiChain NativeChain node managed in docker
+ */
+export class NativeChainContainer extends GenericContainer {
+  /**
+   * @param {string} image docker image name
+   */
+  constructor (
+    image: string = NativeChainContainer.image
+  ) {
+    super(image)
+  }
+
+  static get image (): string {
+    if (process?.env?.DEFICHAIN_DOCKER_IMAGE !== undefined) {
+      return process.env.DEFICHAIN_DOCKER_IMAGE
+    }
+    return 'defi/defichain:HEAD-02ef6a1b3'
+  }
+
+  public static readonly DefaultStartOptions = {
+    user: 'testcontainers-user',
+    password: 'testcontainers-password'
+  }
+
+  protected startOptions: StartOptions = NativeChainContainer.DefaultStartOptions
+
+  public withStartOptions (startOptions: StartOptions): this {
+    this.startOptions = startOptions
+    return this
+  }
+
+  protected networkConfig: JellyfishNetwork = getNetwork('testnet')
+
+  public withJellyfishNetwork (networkName: NetworkName): this {
+    this.networkConfig = getNetwork(networkName)
+    return this
+  }
+
+  protected cachedRpcUrl?: string
+
+  /**
+   * Convenience Cmd builder with StartOptions
+   */
+  protected getCmd (): string[] {
+    const { user, password } = this.startOptions
+
+    return [
+      'defid',
+      '-printtoconsole',
+      '-rpcallowip=0.0.0.0/0',
+      '-rpcbind=0.0.0.0',
+      '-rpcworkqueue=512',
+      `-rpcuser=${user}`,
+      `-rpcpassword=${password}`
+    ]
+  }
+
+  /* eslint-disable @typescript-eslint/no-non-null-assertion, no-void */
+  public static readonly PREFIX = 'defichain-testcontainers-'
+
+  /**
+   * Create container and start it immediately waiting for NativeChain to be ready
+   */
+  public async start (): Promise<StartedNativeChainContainer> {
+    const network = await new Network().start()
```

Comment:
The network configuration should be injected in with `withNetworkMode`, we shouldn't internalize this; it would limit the usability of testcontainer Fluent API and make it impossible to connect with other docker containers using the same docker network.

Reply:
Yes, I was considering this as well, but I was following the `MetaChain` implementation here: https://github.com/DeFiCh/metachain/blob/main/packages/testcontainers/src/MetaChainContainer.ts#L39

I'll change it.

Reply:
Yeah, the implementation there is flawed. Ought to change it there too. Let me open an issue there.